packages/python/plotly/plotly/express/trendline_functions/__init__.py

```diff
@@ -110,11 +111,22 @@ def lowess(trendline_options, x_raw, x, y, x_label, y_label, non_missing):
 
 
 def _pandas(mode, trendline_options, x_raw, y, non_missing):
+    import numpy as np
+
+    try:
+        import pandas as pd
+    except ImportError:
+        msg = "Trendline requires pandas to be installed"
+        raise ImportError(msg)
+
     modes = dict(rolling="Rolling", ewm="Exponentially Weighted", expanding="Expanding")
     trendline_options = trendline_options.copy()
     function_name = trendline_options.pop("function", "mean")
     function_args = trendline_options.pop("function_args", dict())
-    series = pd.Series(y, index=x_raw)
+
+    series = pd.Series(np.copy(y), index=x_raw.to_pandas())
```

Comment:
@MarcoGorelli for old numpy versions, arrow arrays and polars series `to_numpy` seem to be raise 

> ValueError: buffer source array is read-only

when then it is used by pandas operations.

I wonder if we should allow `to_numpy(..., writable: bool)` in narwhals to return a  writable copy

Reply:
For my own curiosity why do we need Pandas still here? Does Narwhals provide a `series` equivalent?

Reply:
We do not currently support `rolling`, `expanding` nor `ewm` at the current stage (see next line TODO comment).

If and when that will be the case, then pandas can be completely optional.
Other way could be to implement these function in pure numpy, but I would rather keep it out of this PR even if that's an option. WDYT?

Reply:
I don't think there's a way to implement rolling functions in numpy, especially with the extra trendline options

We'll add them to Narwhals though, we'd talked about it last month but decided that at the time it wasn't the top priority - but it is on the roadmap

Reply:
Got it. This approach makes sense to me -- I opened https://github.com/plotly/plotly.py/issues/4834 to keep track of this as a follow-up.

Reply:
> I wonder if we should can allow to_numpy(..., writable: bool) in narwhals to return a writable copy

rather than an option, does it makes sense here to check `y.flags['WRITEABLE']`, and copy if it's false?